packages/flutter/lib/src/material/dropdown_menu.dart

```diff
@@ -428,7 +495,7 @@ class _DropdownMenuState<T> extends State<DropdownMenu<T>> {
 
   @override
   void dispose() {
-    _textEditingController.dispose();
+    _textFocusNode.dispose();
```

Comment:
Will update `didUpdateWidget` and tests. Thanks a lot!

But for this condition,
> If requestFocusOnTap is false, then we probably don't need to create a focus node.

Currently, if the `requestFocusOnTap` is false, I use the `focusNode` to unfocus (`_textFocusNode.unfocus();`) the text field when it is tapped. Is this the correct way to keep the `TextField` unfocused? It looks like we don't need to create a focus node when `requestFocusOnTap` is true because the text field will automatically request focus when it's tapped.

Reply:
You probably shouldn't unfocus the text field like that, since there are edge cases where you'll have it flashing as focused for a frame, or having to unfocus at times when you didn't expect to (e.g. when popping a route).

The long term answer is probably to use a `Text` widget instead of a TextField when it is not focusable.  That is much easier said than done, however, since the `Text` widget doesn't have an `InputDecorator`, and we probably want the decoration, and there are a lot of things that the text field does for you here.

~~In the meantime, though, would setting the `readOnly` attribute on the text field be sufficient?~~ Probably not: seems like that's kind of what you want, but I think that it ignores taps then, which might not be ideal.

~~Another alternative would be to always supply a focus node to the text field, and set `canRequestFocus` on the node to be false.~~ Ugh.  No, that won't work, because it is overwritten by the text field, and it only turns it off if the field is disabled.

Okay, how about adding an optional `canRequestFocus` to the text field, and having that override the text field's internal logic if it is set. Then you can pass "false" for that attribute to the text field, and it won't be allowed to accept focus.

Reply:
This sounds good to me! Will make the change. Thanks so much:)